core/src/test/java/feast/core/service/SpecServiceTest.java

```diff
@@ -661,10 +675,20 @@ public void shouldDoNothingIfNoChange() throws InvalidProtocolBufferException {
   }
 
   @Test
-  public void shouldFailIfGetFeatureSetWithoutProject() throws InvalidProtocolBufferException {
-    expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("No project provided");
-    specService.getFeatureSet(GetFeatureSetRequest.newBuilder().setName("f1").build());
+  public void getOrListFeatureSetShouldUseDefaultProjectIfProjectUnspecified()
+      throws InvalidProtocolBufferException {
+    when(featureSetRepository.findFeatureSetByNameAndProject_Name("f4", Project.DEFAULT_NAME))
```

Comment:
Should we be testing the list/find/get/apply without providing the `project` at all? Not sure if I am misreading this, but it seems like you are providing it.

Reply:
Notice that I'm not providing the default project when I'm building list/get/apply request (ie no calls to `.setProject()`). Here I'm only provide the default project to mocks to verify that list/get/apply calls will map to the default project in SpecService and correspondingly makes the correct calls to  the`featureSetRepository` mock with the correct project: `default`

Reply:
Alright, my bad :)